cranelift-codegen/src/regalloc/spilling.rs

```diff
@@ -498,6 +541,13 @@ impl<'a> Context<'a> {
             .filter_map(|lv| {
                 // Viable candidates are registers in one of the `mask` classes, and not already in
                 // the spill set.
+                if let Affinity::RegUnit(unit) = lv.affinity {
+                    let rc = toprc_containing_regunit(unit, &self.reginfo);
+                    if (mask & (1 << rc.toprc)) != 0 && !self.spills.contains(&lv.value) {
+                        // Here, `lv` is a viable spill candidate.
+                        return Some(lv.value);
+                    }
+                }
                 if let Affinity::Reg(rci) = lv.affinity {
```

Comment:
.. this should be an `else if let` but doesn't _break_ anything here. I'll fix that anyway.